osquery/main/windows/daemon.cpp

```diff
@@ -70,6 +75,90 @@ void DebugPrintf(const char* fmt, ...) {
   va_end(vl);
 }
 
+/*
+ * Parses arguments for the Windows service. Arguments to the Windows service
+ * can be passed in two ways: via sc.exe (manual start) or via binPath
+ * (automated start). Unfortunately, both use different methods of getting the
+ * command line arguments. Manual start uses the argc and argv provided by
+ * ServiceMain whereas automated start requires manual parsing of
+ * GetCommandLine()
+ */
+class ServiceArgumentParser {
+ public:
+  ServiceArgumentParser(DWORD argc, LPSTR* argv) {
+    if (argc > 1) {
+      for (DWORD i = 0; i < argc; i++) {
+        args_.push_back(argv[i]);
+      }
+    } else {
+      int wargc = 0;
+      LPWSTR* wargv = ::CommandLineToArgvW(::GetCommandLineW(), &wargc);
+
+      if (wargv != nullptr) {
+        for (int i = 0; i < wargc; i++) {
+          LPSTR arg = toMBString(wargv[i]);
+
+          // On error, bail out and clean up the vector
+          if (arg == nullptr) {
+            cleanArgs();
+            ::LocalFree(wargv);
+            break;
+          }
+          args_.push_back(arg);
```

Comment:
You definitely want to `delete[]` these in the dtor, but otherwise when passed `argv`, you may break things.